### PR TITLE
play kube: add support for configmap binaryData

### PR DESCRIPTION
**Pull request by SoMuchForSubtlety**
This is a breaking change to `KubeVolume`. I first implemented it with a separate `BinaryItems` field, but I thought it's tidier like this.

#### Does this PR introduce a user-facing change?

```release-note
Added support for the `binaryData` field in configmaps for play kube
```

### CODE REVIEW

File: pkg/specgen/generate/kube/volume.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ type KubeVolume struct {
 	// This is only used when there are volumes in the yaml that refer to a configmap
 	// Example: if configmap has data "SPECIAL_LEVEL: very" then the file name is "SPECIAL_LEVEL" and the
 	// data in that file is "very".
-	Items map[string]string
+	Items map[string][]byte
```
Comment (vrothberg) [inline pkg/specgen/generate/kube/volume.go:48]: @Luap99 can you double check whether this is a breaking change?

I don't think so, but I have missed some in the past, so I prefer a head nod from you.

**Review by vrothberg — COMMENTED** [review]
Thanks!

Can you also update docs/kubernetes_support.md to reflect that it's now supported?

File: pkg/specgen/generate/kube/volume.go (hunk 1 above)
Comment (SoMuchForSubtlety) [reply pkg/specgen/generate/kube/volume.go:48]: If anyone is using this package as a library and accesses this field, this is a breaking change. I highly doubt that anyone is actually doing this, and [I can't find any occurrences in public repositories](https://sourcegraph.com/search?q=context:global+%22github.com/containers/podman/v4/pkg/specgen/generate/kube%22&patternType=standard).

**Comment by SoMuchForSubtlety** [comment]
> Can you also update docs/kubernetes_support.md to reflect that it's now supported?

Done :slightly_smiling_face:

File: pkg/specgen/generate/kube/volume.go [hunk 2]
```diff
@@ -207,11 +210,16 @@ func VolumeFromConfigMap(configMapVolumeSource *v1.ConfigMapVolumeSource, config
...
+				kv.Items[item.Path] = []byte(val)
+			} else if val, ok := configMap.BinaryData[item.Key]; ok {
 				kv.Items[item.Path] = val
 			}
 		}
 	} else {
 		for k, v := range configMap.Data {
+			kv.Items[k] = []byte(v)
+		}
+		for k, v := range configMap.BinaryData {
```
[the remark this answers is not in the document]
Comment (SoMuchForSubtlety) [reply pkg/specgen/generate/kube/volume.go:222]: > The keys stored in BinaryData must not overlap with the ones in the Data field, this is enforced during validation process.[^1]

I'll adjust my code accordingly.

[^1]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#configmap-v1-core

**Review by vrothberg — COMMENTED** [review]
LGTM
@Luap99 PTAL

**Review by Luap99 — COMMENTED** [review]
/lgtm

**Comment by rhatdan** [comment]
Thanks @SoMuchForSubtlety

**Pull request merged**